### PR TITLE
issue-1149

### PR DESCRIPTION
**Pull request by Spoilsport0**
## Summary of issue
When the 'imagesDetails' are deleted from the request body, the catalog on the frontend fails

## Summary of change

set response 400 if we have one of the following situations:
 without "alt" string in the "imagesDetails"
 with "imagesDetails": null
 without all massif "imagesDetails"
 with invalid "imagesIds": [] )

### CODE REVIEW

File: Streetcode/Streetcode.BLL/MediatR/Streetcode/Streetcode/Create/CreateStreetcodeHandler.cs [hunk 1]
```diff
@@ -151,9 +157,22 @@ await _repositoryWrapper.StreetcodeImageRepository.CreateRangeAsync(imagesIds.Se
...
+        if(imageDetails.IsNullOrEmpty())
+        {
+            throw new HttpRequestException("There is no valid imagesDetails value", null, System.Net.HttpStatusCode.BadRequest);
+        }
+
         if(imageDetails == null)
         {
-            return;
+            throw new HttpRequestException("There is no valid imagesDetails value", null, System.Net.HttpStatusCode.BadRequest);
+        }
```
Comment (yaakymiv) [inline Streetcode/Streetcode.BLL/MediatR/Streetcode/Streetcode/Create/CreateStreetcodeHandler.cs:168]: These parts seem to be interchangeable. Do you need both of them?

Comment (yaakymiv) [inline Streetcode/Streetcode.BLL/MediatR/Streetcode/Streetcode/Create/CreateStreetcodeHandler.cs:168]: These parts seem to be interchangeable . Do you need both of them?

Comment (yaakymiv) [inline Streetcode/Streetcode.BLL/MediatR/Streetcode/Streetcode/Create/CreateStreetcodeHandler.cs:168]: These parts seem to be interchangeable. Do you need both of them?

Comment (Spoilsport0) [reply Streetcode/Streetcode.BLL/MediatR/Streetcode/Streetcode/Create/CreateStreetcodeHandler.cs:168]: you are right, it works exactly good without the second check, I will remove it

Comment (Lazy-Lenny) [inline Streetcode/Streetcode.BLL/MediatR/Streetcode/Streetcode/Create/CreateStreetcodeHandler.cs]: variable.IsNullOrEmpty() is more versatile and commonly used when you want to check if the variable is null or empty, regardless of its type, whereas var == null is specifically for checking if a variable is null. You basically making additional check that never be neither completed or true. Delete second statement

File: Streetcode/Streetcode.BLL/MediatR/Streetcode/Streetcode/Create/CreateStreetcodeHandler.cs [hunk 2]
```diff
@@ -151,9 +157,22 @@ await _repositoryWrapper.StreetcodeImageRepository.CreateRangeAsync(imagesIds.Se
...
+
+        foreach (var detail in imageDetails)
+        {
+            if(detail.Alt == null)
+            {
+                throw new HttpRequestException("There is no valid imagesDetails value", null, System.Net.HttpStatusCode.BadRequest);
+            }
```
Comment (Lazy-Lenny) [inline Streetcode/Streetcode.BLL/MediatR/Streetcode/Streetcode/Create/CreateStreetcodeHandler.cs:170]: Assuming that the 'Alt' property is a string, why would we use detail.Alt == null, when we should use detail.Alt.isNullOrEmpty? Can 'Alt' property be an empty string?

**Review by Lazy-Lenny — CHANGES_REQUESTED** [review]

File: Streetcode/Streetcode.BLL/MediatR/Streetcode/Streetcode/Create/CreateStreetcodeHandler.cs (hunk 1 above)
Comment (Spoilsport0) [reply Streetcode/Streetcode.BLL/MediatR/Streetcode/Streetcode/Create/CreateStreetcodeHandler.cs]: I have already removed second statement

**Review by Lazy-Lenny — APPROVED** [review]

**Review by yaakymiv — APPROVED** [review]

**Review by Lazy-Lenny — APPROVED** [review]

**Review by vasylashka — APPROVED** [review]

**Pull request merged**